core-5/src/main/java/app/config/ProjectConfig.java

```diff
@@ -0,0 +1,74 @@
+package app.config;
+
+import app.car.*;
+import app.engine.Engine;
+import app.engine.FourCylinderEngine;
+import app.engine.V12Engine;
+import app.engine.V6Engine;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.stereotype.Component;
+
+@Component
+@PropertySource("classpath:context.properties")
+public class ProjectConfig {
+
+    @Value("${ford.engine.cylinders}")
```

Comment:
okay, looks good, but you can do better

right now these fields are available to be read by other beans, lets try to find a way to make it visible only for beans that are supposed to use them